overlord/servicestate/quota_handlers.go

```diff
@@ -221,6 +223,94 @@ func addRestartServicesTasks(st *state.State, queueTask func(task *state.Task),
 	}
 }
 
+func AddSnapToQuotaGroup(st *state.State, snapName string, quotaGroup string) (*state.Task, error) {
+	if err := CheckQuotaChangeConflictMany(st, []string{quotaGroup}); err != nil {
+		return nil, err
+	}
+
+	// This could result in doing 'setup-profiles' twice, but
+	// unfortunately we can't execute this code earlier as the snap
+	// needs to appear as installed first.
+	quotaControlTask := st.NewTask("quota-add-snap", fmt.Sprintf(i18n.G("Add snap %q to quota group %q"),
+		snapName, quotaGroup))
+	quotaControlTask.Set("snap-name", snapName)
```

Comment:
given that snapstate will also attach a SnapSetup to the task, it's not clear we need this bit, we could use snapstate.TaskSnapSetup and get the snap name from that

Reply:
I changed the task to use the attached SnapSetup, this worked nicely, thank you for the suggestion. The function still takes the snap name as a parameter so we can add it to the summary of the task.